### PR TITLE
Feat/guide

### PR DESCRIPTION
**Pull request by cythb**
- add a guide scene
- status bar appearance based on view controller

### CODE REVIEW

File: Maskbook/Scene/Wallet/Guide/GuideView.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+//  GuideView.swift
+//  Maskbook
+//
+//  Created by Hugo L on 2022/2/25.
+//  Copyright © 2022 dimension. All rights reserved.
+//
+
+import SwiftUI
+
+struct GuideView: View {
```
Comment (foxsin10) [inline Maskbook/Scene/Wallet/Guide/GuideView.swift:11]: By adding `Hashable` to `GuideItemView.Page`, the following code will  fit the demand.
If we need a `PageControl`, the page property which has been attributed by `@State` will help.

```
struct GuideView: View {
    @State
    private var page = GuideItemView.Page.one

    var body: some View {
        TabView(selection: $page) {
            ForEach(GuideItemView.Page.allCases, id: \.self) { page in
                GuideItemView(page: page)
                    .tag(page)
            }
        }
        .tabViewStyle(.page(indexDisplayMode: .never))
        .withBlueGradient(radius: 0, colorDecorator: .guideGradient)
        .ignoresSafeArea()
        .colorScheme(.light)
    }
}
```

File: Maskbook/Scene/MainTab/MainTabBarController.swift [hunk 2]
```diff
@@ -81,14 +81,6 @@ class MainTabBarController: UITabBarController {
 }
 
 extension MainTabBarController {
-    override var childForStatusBarStyle: UIViewController? {
```
Comment (foxsin10) [inline Maskbook/Scene/MainTab/MainTabBarController.swift:84]: Why delete this？

**Comment by foxsin10** [comment]
We don't need `Pages`.

File: Maskbook/Scene/Wallet/Guide/GuideItemView.swift [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    }
+    
+    var pageControlSection: some View {
+        VStack(alignment: .center) {
+            page.pageControl.resizable().frame(width: 71, height: 6)
+        }
+    }
+    
+    var topSpace: CGFloat {
+        0.114_532_02 * UIScreen.main.bounds.height
```
Comment (sxiaojian88) [inline Maskbook/Scene/Wallet/Guide/GuideItemView.swift:43]: I can't understand this value, I think this format **/**  is better

File: Maskbook/Scene/MainTab/MainTabBarController.swift (hunk 2 above)
Comment (cythb) [reply Maskbook/Scene/MainTab/MainTabBarController.swift:84]: This method has moved into `NavigationController`.

File: Maskbook/Scene/Guide/PagerView.swift [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    // swiftlint:enable implicitly_unwrapped_optional
+    
+    func updateUIView(_ uiView: UIScrollView, context: Context) {
+    }
+    
+    func makeCoordinator() -> Coordinator {
+        Coordinator(parent: self)
+    }
+    
+    class Coordinator: NSObject, UIScrollViewDelegate {
```
Comment (sxiaojian88) [inline Maskbook/Scene/Guide/PagerView.swift:60]: Same name with the router `Coordinator`

**Review by sxiaojian88 — APPROVED** [review]

**Review by foxsin10 — APPROVED** [review]

File: Maskbook/Supporting Files/Coordinator.swift [hunk 5]
```diff
@@ -321,6 +328,12 @@ extension Coordinator {
         case .persona:
             return PersonasViewController()
 
+        case .guide:
+            let vc = MaskHostViewController(rootView: AnyView(EmptyView()))
```
[the remark this answers is not in the document]
Comment (cythb) [reply Maskbook/Supporting Files/Coordinator.swift]: `MaskHostViewController` provide a `preferredStatusBarStyle` for showing status bar in light content style. `UIHostingController` can not control the style of status bar

**Pull request merged**